domains/client/relayer/src/lib.rs

```diff
@@ -476,31 +479,75 @@ where
         .map_err(Error::UnableToSubmitCrossDomainMessage)
     }
 
-    fn last_relayed_block_key(domain_id: DomainId) -> Vec<u8> {
-        (b"message_relayer_last_processed_block_of_domain", domain_id).encode()
+    fn relayed_blocks_at_number_key(domain_id: DomainId, number: NumberFor<Block>) -> Vec<u8> {
+        (
+            b"message_relayer_processed_block_of_domain",
+            domain_id,
+            number,
+        )
+            .encode()
     }
 
-    fn fetch_last_relayed_block(
+    /// Takes number as tip and finds all the unprocessed blocks including the tip.
+    fn fetch_unprocessed_blocks_until(
         client: &Arc<Client>,
         domain_id: DomainId,
-    ) -> Option<NumberFor<Block>> {
-        let encoded = client
-            .get_aux(&Self::last_relayed_block_key(domain_id))
-            .ok()??;
+        number: NumberFor<Block>,
+        hash: Block::Hash,
```

Comment:
Would this be clearer?:
```suggestion
        best_number: NumberFor<Block>,
        best_hash: Block::Hash,
```

Reply:
I initially went with it but realised its not always the best we will be seeing. Hence reverted to just number and hash

Reply:
I was considering it from local context point of view: as far as this function is concerned we're going from some best block deeper into the history. Can be left as is too, I don't have strong confidence it'll make things better.